src/hal/src/queue/mod.rs

```diff
@@ -70,6 +71,59 @@ pub struct Submission<Ic, Iw, Is> {
     pub signal_semaphores: Is,
 }
 
+/// Information for binding sparse memory using a [command queue][CommandQueue].
+///
+/// The info is sent to the device through the [`bind_sparse`][CommandQueue::bind_sparse] method.
+#[derive(Debug)]
+pub struct BindSparseInfo<Iw, Is, Ib, Io, Ii> {
+    /// Semaphores to wait being signalled before submission.
+    pub wait_semaphores: Iw,
+    /// Semaphores to signal after all command buffers in the submission have finished execution.
+    pub signal_semaphores: Is,
+    /// Defines sparse buffer memory bind operations.
+    pub buffer_memory_binds: Ib,
+    /// Defines sparse image opaque memory bind operations.
+    pub image_opaque_memory_binds: Io,
+    /// Defines sparse image memory bind operations.
+    pub image_memory_binds: Ii,
+}
+
+/// Defines a single memory bind region.
+///
+/// This is used in the [`bind_sparse`][CommandQueue::bind_sparse] method to define a physical
+/// store region for a buffer.
+#[derive(Debug)]
+pub struct SparseMemoryBind<M> {
+    /// Offset into the (virtual) resource.
+    pub resource_offset: u64,
+    /// Size of the memory region to be bound.
+    pub size: u64,
+    /// Memory that the physical store is bound to, and the offset into the resource of the binding.
+    ///
+    /// Using `None` will unbind this range. Reading or writing to an unbound range is undefined
+    /// behaviour in some older hardware.
+    pub memory: Option<(M, usize)>,
+}
+
+/// Defines a single image memory bind region.
+///
+/// This is used in the [`bind_sparse`][CommandQueue::bind_sparse] method to define a physical
+/// store region for a buffer.
+#[derive(Debug)]
+pub struct SparseImageMemoryBind<'a, M> {
```

Comment:
I wonder if these structs should instead live in `memory` module. They can be named without the `Memory` suffix in them in this case, like `SparseImageBind`